apps/scotty/collect.go

```diff
@@ -280,12 +280,12 @@ type memoryCheckerType interface {
 	Check()
 }
 
-func createCloudHealthWriter(path string) *cloudhealth.Writer {
+func newCloudHealthWriter(reader io.Reader) (interface{}, error) {
```

Comment:
Can you define a more specific interface type with the appropriate method signatures? The empty interface type is basically a void pointer and is discouraged.